### ISSUE TITLE
TUnit0023 rule incorrect when using primary constructors

### ISSUE THREAD

**Issue by abc-rpg** [thomhurst/TUnit#1396]
### Issue

When a test class uses a primary constructor, TUnit0023 shows warnings about disposable objects that _are_ being cleaned up.

### Example

```csharp
public class Test() // note the use of primary constructor here
{
    private HttpClient _client = null!;

    [Before(HookType.Test)]
    public void Setup()
    {
        _client = new HttpClient();
    }

    [After(HookType.Test)]
    public void Cleanup()
    {
        _client.Dispose();
    }

    [Test]
    public async Task TestMethod()
    {
        var result = true;
        await Assert.That(result).IsTrue();
    }
}
```

Running `dotnet build` gives the following warning:
> warning TUnit0023: _client should be disposed within a clean up method

The warning goes away when removing the primary constructor.

### Notes
* The `HttpClient` here is just an example of an `IDisposable`
* The primary constructor is empty just to showcase the minimum setup for reproducing the issue. I'm typically passing in the data coming from the ClassDataSource.
* Using a "traditional" constructor does not cause the issue.

### Versions
* dotnet 9
* TUnit 0.4.99

**Comment by thomhurst** [comment]
Thanks for the report

**Comment by thomhurst** [comment]
Should be fixed in v0.4.105

**Comment by abc-rpg** [comment]
Thanks for the quick response! The fix is verified 👍
